pkg/controller/spinnakerservice/spinnakerservice_controller.go

```diff
@@ -21,30 +24,32 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logf.Log.WithName("spinnakerservice")
-
-var SpinnakerServiceBuilder spinnakerv1alpha2.SpinnakerServiceBuilderInterface
+var (
+	log                     = logf.Log.WithName("spinnakerservice")
+	SpinnakerServiceBuilder spinnakerv1alpha2.SpinnakerServiceBuilderInterface
+	DeployerGenerators      = []deployerGenerator{spindeploy.NewDeployer}
+)
 
 // Add creates a new SpinnakerService Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
 	return add(mgr, newReconciler(mgr))
 }
 
-type deployer interface {
-	IsSpinnakerUpToDate(ctx context.Context, svc spinnakerv1alpha2.SpinnakerServiceInterface) (bool, error)
-	Deploy(ctx context.Context, svc spinnakerv1alpha2.SpinnakerServiceInterface, scheme *runtime.Scheme) error
-}
+type deployerGenerator func(m deploy.ManifestGenerator, mgr manager.Manager, clientset *kubernetes.Clientset, logger logr.Logger) deploy.Deployer
```

Comment:
Could this be moved to `deploy` module, so that both deployer and its generator live in the same place?

Reply:
We don't need it in the other module. I'd argue the generator per se is only used here. in `deploy/spindeploy` we only have a regular `NewXXX` function